### PR TITLE
Allow proxy-only and composed virtual workspaces 

### PR DESCRIPTION
**Pull request by davidfestal**

### CODE REVIEW

File: pkg/virtual/framework/handlerbased/types.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/server/healthz"
+
+	"github.com/kcp-dev/kcp/pkg/virtual/framework"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+)
+
+var _ framework.VirtualWorkspace = (*HandlerBasedVirtualWorkspace)(nil)
+
+type HandlerBasedVirtualWorkspace struct {
```
Comment (sttts) [inline pkg/virtual/framework/handlerbased/types.go:32]: docs

File: pkg/virtual/framework/chained/types.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/server/healthz"
+
+	"github.com/kcp-dev/kcp/pkg/virtual/framework"
+	virtualcontext "github.com/kcp-dev/kcp/pkg/virtual/framework/context"
+)
+
+var _ framework.VirtualWorkspace = (*ChainedVirtualWorkspace)(nil)
+
+type ChainedVirtualWorkspace struct {
```
Comment (sttts) [inline pkg/virtual/framework/chained/types.go:33]: docs

File: pkg/virtual/framework/types.go [hunk 3]
```diff
@@ -18,11 +18,25 @@ package framework
...
 import (
 	"context"
+	"net/http"
 
 	"k8s.io/apiserver/pkg/authorization/authorizer"
 	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/apiserver/pkg/server/healthz"
 )
 
+type VirtualWorkspaceName string
```
Comment (sttts) [inline pkg/virtual/framework/types.go:28]: what is this name? Which format? Where used?

**Comment by stevekuznetsov** [comment]
/hold

I have a copy of this in my branch, likely best to merge that instead of reconciling.

**Comment by ncdc** [comment]
@davidfestal @stevekuznetsov what is the status on this - still want it, or close?

**Pull request closed without merging**